packages/design-system/src/components/OcResourceIcon/OcResourceIcon.vue

```diff
@@ -41,30 +44,43 @@ export default {
     size: {
       type: String,
       default: 'large',
-      validator: (value) => {
-        return value.match(/(xsmall|small|medium|large|xlarge|xxlarge|xxxlarge)/)
+      validator: (value: string): boolean => {
+        return AVAILABLE_SIZES.some((e) => e === value)
       }
     }
   },
   computed: {
     iconName() {
+      if (this.isSpace) return defaultSpaceIcon
       if (this.isFolder) return defaultFolderIcon
       const icon = iconNameMap[this.extension]
       return `resource-type-${icon ? icon : defaultFallbackIcon}`
     },
     iconColor() {
+      if (this.isSpace) return defaultSpaceColor
       if (this.isFolder) return defaultFolderColor
       const color = iconColorMap[this.extension]
       return color ? color : defaultFallbackIconColor
     },
+    iconTypeClass() {
+      if (this.isSpace) return 'oc-resource-icon-space'
+      if (this.isFolder) return 'oc-resource-icon-folder'
+      return 'oc-resource-icon-file'
+    },
     isFolder() {
-      return this.resource.isFolder
+      // fallback is necessary since
+      // sometimes resources without a type
+      // but with `isFolder` are being passed
+      return this.resource.type === 'folder' || this.resource.isFolder
```

Comment:
The class checks above only check for `if (this.isFolder) {...}`. Can we omit the `type` check here as well and always use the "fallback"?

Reply:
`if (this.isFolder) return 'oc-resource-icon-folder'` => checks the computed property `isFolder`
while
`return this.resource.type === 'folder' || this.resource.isFolder` checks the resource passed as prop for its `isFolder` attribute ;)

Ideally, any resource passed to the OcResourceIcon has a `type`, but in some cases it only has a `isFolder` (and I'm not too sure how official from an api-standpoint the `isFolder` resource attribute is anyhow)...just checked and there's some usage of `isFolder` across the conflict&sharing helpers, actions mixins and upload components so most likely out of scope for this PR

Reply:
Ahh sorry, didn't watch closely enough. Yeah, it would be great to streamline that, but definitely out of scope here :D

Reply:
I get it though, I had fun in this component. Things don't get better as a space also has a `isFolder === true` while a resource-folder apparently can potentially have either `isFolder` or `type` of undefined  🥴 riddle me this